.github/workflows/build-and-test.yml

```diff
@@ -123,6 +123,55 @@ jobs:
         with:
           os: ${{matrix.os}}
 
+  build-and-test-stardust:
+    needs: [ check-for-run-condition, check-for-modification ]
+    if: ${{ needs.check-for-run-condition.outputs.should-run == 'true' && needs.check-for-modification.outputs.core-modified == 'true' }}
```

Comment:
can / should we get more elegant with the modification detection or is it too interconnected?

Reply:
It relies on the core libraries. It would be too fine-grained to detect changes to core libraries versus IOTA-legacy libraries vs Stardust library.